index.js

```diff
@@ -0,0 +1,133 @@
+const core = require("@actions/core");
+const yaml = require("js-yaml");
+const path = require("path");
+const fs = require("fs");
+
+const jsonSchema = {
+  $schema: "http://json-schema.org/draft-07/schema#",
+  type: "object",
+  properties: {},
+  required: [],
+};
+
+function readInput() {
+  try {
+    sourceRootDir = core.getInput("source-root-dir-path");
+    return sourceRootDir;
+  } catch (error) {
+    throw new Error(`Failed to read input: ${error.message}`);
+  }
+}
+
+function readComponentYaml(filePath) {
+  try {
+    fullPath = path.join(filePath, ".choreo", "component.yaml");
+    let fileContent = fs.readFileSync(fullPath, "utf8");
+    return fileContent;
+  } catch (error) {
+    throw new Error(`Failed to read component.yaml: ${error.message}`);
+  }
+}
+
+function generateSchemaFromYaml(schema, requiredItems) {
+  if (schema.type === "string") {
+    if (!typeof schema.required === "boolean" || schema.required !== false) {
```

Comment:
Shall we add comment explaining why we are doing this, it isn't clear. Also would it be better to divide these logics into sub fn given that this could expand further in the future, WDYT?

Reply:
Fixed the review comments

Reply:
https://github.com/choreo-templates/choreo-config-schema-generator/pull/1/commits/e0ff6715161e444e55cb50820859ba39770028a2